Assets/VRTK/SDK/Oculus/SDK_OculusController.cs

```diff
@@ -728,21 +740,52 @@ protected virtual bool IsButtonPressed(uint index, ButtonPressTypes type, OVRInp
             VRTK_TrackedController device = GetTrackedObject(GetControllerByIndex(index));
             if (device != null)
             {
-                OVRInput.Controller controller = touchControllers[index];
+                OVRInput.Controller controllerMask = GetControllerMask(device.index);
                 switch (type)
                 {
                     case ButtonPressTypes.Touch:
-                        return OVRInput.Get(button, controller);
+                        return OVRInput.Get(button, controllerMask);
                     case ButtonPressTypes.TouchDown:
-                        return OVRInput.GetDown(button, controller);
+                        return OVRInput.GetDown(button, controllerMask);
                     case ButtonPressTypes.TouchUp:
-                        return OVRInput.GetUp(button, controller);
+                        return OVRInput.GetUp(button, controllerMask);
                 }
             }
 
             return false;
         }
 
+        protected virtual OVRInput.Controller GetControllerMask(uint index)
+        {
+            if (GetCurrentControllerType() == ControllerType.Oculus_OculusTouch)
+            {
+                return touchControllers[index];
+            }
+            return OVRInput.GetActiveController();
+        }
+
+        protected virtual OVRInput.RawAxis2D GetTouchpadAxisMask(uint index)
+        {
+            switch (GetCurrentControllerType())
+            {
+                case ControllerType.Oculus_OculusTouch:
+                case ControllerType.Oculus_OculusGamepad:
+                    return thumbsticks[index];
+            }
+            return touchpads[index];
+        }
+
+        protected virtual OVRInput.Button GetTouchpadButtonMask()
```

Comment:
Button mask needs to take the touch type, at the moment it only returns Button press and does not return Touch state